locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/AbstractComputeServiceRegistry.java

```diff
@@ -0,0 +1,283 @@
+                }
+                properties.setProperty(LocationConstants.PROPERTY_REGIONS, region);
+            }
+
+            // occasionally can get com.google.common.util.concurrent.UncheckedExecutionException: java.lang.RuntimeException:
+            //     security group eu-central-1/jclouds#brooklyn-bxza-alex-eu-central-shoul-u2jy-nginx-ielm is not available after creating
+            // the default timeout was 500ms so let's raise it in case that helps
+            properties.setProperty(EC2Constants.PROPERTY_EC2_TIMEOUT_SECURITYGROUP_PRESENT, "" + Duration.seconds(30).toMilliseconds());
+            return this;
+        }
+
+        private PropertiesBuilder setAzureComputeArmProperties() {
+            String region = conf.get(CLOUD_REGION_ID);
+            if (Strings.isNonBlank(region)) {
+                properties.setProperty(LocationConstants.PROPERTY_REGIONS, region);
+            }
+            return this;
+        }
+
+        private PropertiesBuilder setJCloudsProperties() {
+            // Add extra jclouds-specific configuration
+            Map<String, Object> extra = Maps.filterKeys(conf.getAllConfig(), Predicates.containsPattern("^jclouds\\."));
+            if (extra.size() > 0) {
+                String provider = getProviderFromConfig(conf);
+                LOG.debug("Configuring custom jclouds property overrides for {}: {}", provider, Sanitizer.sanitize(extra));
+            }
+            properties.putAll(Maps.filterValues(extra, Predicates.notNull()));
+            return this;
+        }
+
+        private PropertiesBuilder setEndpointProperty() {
+            String endpoint = conf.get(CloudLocationConfig.CLOUD_ENDPOINT);
+            if (!groovyTruth(endpoint)) endpoint = getDeprecatedProperty(conf, Constants.PROPERTY_ENDPOINT);
+            if (groovyTruth(endpoint)) properties.setProperty(Constants.PROPERTY_ENDPOINT, endpoint);
+            return this;
+        }
+
+        public Properties build() {
+            return properties;
+        }
+    }
+
+    public class ComputeServiceSupplier implements Supplier<ComputeService> {
+
+        private final String provider;
+        private final ConfigBag conf;
+        private final Iterable<? extends Module> modules;
+        private final Properties properties;
+
+        private final Object createComputeServicesMutex = new Object();
+
+        public ComputeServiceSupplier(ConfigBag conf, Iterable<? extends Module> modules, Properties properties) {
+            this.provider = getProviderFromConfig(conf);
+            this.conf = conf;
+            this.modules = modules;
+            this.properties = properties;
+        }
+
+        public ComputeService get() {
+            // Synchronizing to avoid deadlock from sun.reflect.annotation.AnnotationType.
+            // See https://github.com/brooklyncentral/brooklyn/issues/974
+            synchronized (createComputeServicesMutex) {
```

Comment:
@aledsage would you mind taking a look at this?  I agree with @andreaturli that it is probably no longer required, but then, I don't really understand the threading model that cause this deadlock in the first place.